src/host/readDataCooked.cpp

```diff
@@ -351,79 +165,30 @@ bool COOKED_READ_DATA::Notify(const WaitTerminationReason TerminationReason,
     // We must see if we were woken up because the handle is being closed. If
     // so, we decrement the read count. If it goes to zero, we wake up the
     // close thread. Otherwise, we wake up any other thread waiting for data.
-
     if (WI_IsFlagSet(TerminationReason, WaitTerminationReason::HandleClosing))
     {
         *pReplyStatus = STATUS_ALERTED;
         gci.SetCookedReadData(nullptr);
         return true;
     }
 
-    // If we get to here, this routine was called either by the input thread
-    // or a write routine. Both of these callers grab the current console
-    // lock.
-
-    // MSFT:13994975 This is REALLY weird.

```

Comment:
Is this scary comment no longer relevant? I remember this as one of the wackiest bits of COOKED_READ we ever debugged. 

Seems like it's okay because we're not stashing `_pdwNumBytes` anywhere... though, are we ever writing to `pNumBytes` now?

Reply:
I don't think this is relevant anymore. The existing cooked read instance now owns the popups and will directly feed them with input.